src/ApiTestCase.php

```diff
@@ -0,0 +1,202 @@
+            $this->client->getContainer()->unmock($id);
+        }
+
+        \Mockery::close();
+        $this->client = null;
+
+        parent::tearDown();
+    }
+
+    /**
+     * Gets service from DIC.
+     *
+     * @param string $id
+     *
+     * @return object
+     */
+    protected function get($id)
+    {
+        return $this->client->getContainer()->get($id);
+    }
+
+    /**
+     * @param Response $response
+     * @param int      $statusCode
+     */
+    protected function assertResponseCode(Response $response, $statusCode)
+    {
+        $this->assertEquals($statusCode, $response->getStatusCode());
+    }
+
+    /**
+     * @param Response $response
+     *
+     * @throws \Exception
+     */
+    protected function assertSuccessfulResponse(Response $response)
+    {
+        $openErrorsInBrowser = false;
+        if (isset($_SERVER['OPEN_ERROR_IN_BROWSER'])) {
+            $openErrorsInBrowser = $_SERVER['OPEN_ERROR_IN_BROWSER'];
+        }
+
+        if (in_array($response->getStatusCode(), [400, 401, 403, 404]) && $openErrorsInBrowser) {
+            $filename = rtrim(sys_get_temp_dir(), DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR.uniqid().'.html';
+            file_put_contents($filename, $response->getBody()->getContents());
+            system(sprintf('open %s', escapeshellarg($filename)));
+
+            throw new \Exception('Internal server error.');
+        }
+    }
+
+    /**
+     * @param Response $response
+     * @param int      $statusCode
+     */
+    protected function assertJsonResponse($response, $statusCode = 200)
+    {
+        $this->assertEquals(
+            $statusCode, $response->getStatusCode(),
+            $response->getContent()
+        );
+
+        $this->assertTrue(
+            $response->headers->contains('Content-Type', 'application/json'),
+            $response->headers
+        );
+    }
+
+    /**
+     * Asserts that response has JSON content matching the one given in file.
+     *
+     * @param Response $response
+     * @param string   $filename
+     *
+     * @throws \Exception
+     */
+    protected function assertJsonResponseContent(Response $response, $filename)
+    {
+        if (!isset($_SERVER['EXPECTED_RESPONSE_DIR'])) {
+            throw new \Exception('You have to set a EXPECTED_RESPONSE_DIR variable.');
+        }
+
+        $expectedResponse = file_get_contents(sprintf(
+            '%s/%s/%s.json',
+            $this->getRootDir(),
+            $_SERVER['EXPECTED_RESPONSE_DIR'],
+            $filename
+        ));
+        $actualResponse = $response->getContent();
+
+        $actualResponse = json_encode(json_decode($actualResponse), JSON_PRETTY_PRINT);
+
+        $factory = new SimpleFactory();
+        $matcher = $factory->createMatcher();
+
+        $result = $matcher->match($actualResponse, $expectedResponse);
+
+        if (!$result) {
+            echo $matcher->getError();
```

Comment:
I am fine with this until we try it out on Sylius, but we should figure out a better way to output these than echo, if possible. I'd have a look in PHPSpec how is it displaying array diffs and errors etc.